xgcm/sgrid.py

```diff
@@ -0,0 +1,230 @@
+from collections import OrderedDict
+
+
+def assert_valid_sgrid(ds):
+    """Verify that the dataset meets SGRID conventions
+    We know this if the dataset has 'SGRID' listed in its conventions
+
+    Parameters
+    ----------
+    ds : xarray.dataset
+
+    Returns
+    ----------
+    True/False : boolean
+    """
+
+    conventions_attr = next(
+        (x for x in ["Conventions", "conventions"] if x in ds.attrs), False
+    )
+    if conventions_attr:
+        if any(x in ds.attrs[conventions_attr] for x in ["SGRID", "sgrid", "Sgrid"]):
+            return True
+
+    return False
+
+
+def get_sgrid_grid(ds):
+    """Extract the name of the variable containing the SGRID grid data.
+    This can be done by checking cf_role attribute for 'grid_topology'
+
+    Parameters
+    ----------
+    ds : xarray.dataset
+
+    Returns
+    ----------
+    var_name : str
+        The name of the grid variable
+    """
+
+    for var_name in ds.data_vars:
```

Comment:
```suggestion
    for var_name in ds.variables:
```

ALternatively 
```
import  cf_xarray

return ds.cf["grid_topology"]
```

we'll  have to make an upstream change though

Reply:
@jbusecke I think this will add cf_xarray as a dependency of xgcm.
Are you happy for me to do this?
I assume you'll probably want it for any future CF parsing code.

Reply:
@dcherian I've tried the cf_array approach, but it doesn't work as I assume the upstream changes haven't made it to pypi yet?
Do you know when they might appear?
I can leave the code as is (with the cf_xarray alternative commented out) with a TODO note to switch once this feature becomes available?

Reply:
Could just ignore it for now. It'll be more  useful once you start parsing CF conventions for metrics like area and volume I think.